### DISCUSSION TITLE
Quarkus 3.13 - OpenTelemetry Metrics, OpenTelemetry 1.39, TLS registry improvements and more...

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#42254] [Quarkus Blog/Website]
# Quarkus 3.13 - OpenTelemetry Metrics, OpenTelemetry 1.39, TLS registry improvements and more...

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/quarkus-3-13-0-released/

**Comment by PedroRodP** [comment]
We're experiencing a different behavior in `@QuarkusTest`s with this update (not happening in 3.12.3 or earlier versions): **each** test class will execute a `@Startup` annotated method on an `@ApplicationScoped` class that we defined for migrating the database while previously it was invoked just once for the whole test set.
Is this intended? We're not talking about a TestResource here, but might have something to do

**Reply by PedroRodP** [reply]
Well. Actually we do have a TestResource. I tried setting restrictToAnnotatedClass to false and worked as before. But I don't understand why that particular resource not being shared makes the whole context to be reset for a given test class.

**Reply by gsmet** [reply]
@geoand might be of interest to you ^

**Reply by geoand** [reply]
Definitely, I'll take a look tomorrow

**Reply by geoand** [reply]
@PedroRodP do you have a sample application that shows the behavior change between 3.12 and 3.13?

**Reply by gsmet** [reply]
@PedroRodP ideally, please create an issue with a small reproducer. There is something fishy going on and we need to track it properly. Thanks!

**Reply by PedroRodP** [reply]
Sure: https://github.com/quarkusio/quarkus/issues/42365

**Comment by DempseyRoller** [comment]
Updating a project from 3.12.3 to 3.13.0, building a docker image and running it on one of our environments works fine but on another environment the program starts to spam: `ERROR [io.ve.ex.we.RoutingContext] (vert.x-eventloop-thread-2) Unhandled exception in router`. Any tips on what could be some causes for this? Both are running on same Linux kernel version. The working environment has docker 26.1.3 and the non-working one has 26.1.2.

**Reply by gsmet** [reply]
You don't have more information about the error? No stacktrace?

**Reply by DempseyRoller** [reply]
No, no stacktrace. It's very uninformative to be honest.
